### PR TITLE
cleanup group switching output to terminal

### PR DESCRIPTION
**Pull request by totto82**
As requested by @OPMUSER  in https://github.com/OPM/opm-simulators/pull/3837#issuecomment-1141853225

I dropped the switching history RATE->FLE->RATE i.e. only the final change is dumped to the terminal. 
The old logging is moved to .DBG so if you are interested in detailed switching history you will find it there. 

I suggest doing the same for the wells as
ORAT->GRAT->ORAT type of output may not be that useful for the user IMO.

### CODE REVIEW

**Comment by totto82** [comment]
jenkins build this please

File: opm/simulators/wells/BlackoilWellModel_impl.hpp [hunk 1]
```diff
@@ -463,6 +465,35 @@ namespace Opm {
...
+                if (to != from) {
+                    local_deferredLogger.info(msg);
+                }
+            }
+            for (const auto& [key, to] : switched_inj_groups_) {
+                const std::string& name = key.first;
+                const Opm::Phase& phase = key.second;
+                std::string msg = "    Injection Group " + name
+                + " control mode changed from ";
+                const Group& group = schedule().getGroup(name, reportStepIdx);
```
Comment (akva2) [inline opm/simulators/wells/BlackoilWellModel_impl.hpp:487]: unused.

**Comment by totto82** [comment]
jenkins build this please

**Comment by totto82** [comment]
jenkins build this please

**Comment by totto82** [comment]
The "skip" logic was changed by the PR -> unintended changes in the results. 
The last commit just remove it completely. 
We may want to do that in a separate PR? So this is for testing.

**Comment by totto82** [comment]
jenkins build this please

**Comment by totto82** [comment]
benchmark please

**Comment by ytelses** [comment]
Benchmark result overview: 




| Test    | Configuration                                                                                 | Relative |
| ------- | --------------------------------------------------------------------------------------------- | -------- |
| opm-git | OPM Benchmark: flow_mpi_extra - Threads: 1                                                    | 1.026    |
| opm-git | OPM Benchmark: flow_mpi_extra - Threads: 8                                                    | 1.003    |
| opm-git | OPM Benchmark: flow_mpi_norne - Threads: 1                                                    | 1.002    |
| opm-git | OPM Benchmark: flow_mpi_norne - Threads: 8                                                    | 0.99     |
| opm-git | OPM Benchmark: flow_mpi_norne_4c_msw - Threads: 1 - FOPT (Total Oil Production At End Of Run) | 1        |
| opm-git | OPM Benchmark: flow_mpi_norne_4c_msw - Threads: 8 - FOPT (Total Oil Production At End Of Run) | 1        |
* Speed-up = Total time master / Total time pull request. Above 1.0 is an improvement. *



View result details @ https://www.ytelses.com/opm/?page=result&id=1761

**Comment by totto82** [comment]
The "Skip" logic that this PR removes was needed earlier to "compensate" for bugs in the switching code. Since the bugs are long gone the logic is not needed anymore as also the results from the benchmark shows. I hope this can be merged as is without first making a separate PR that removes the logic and then improve on the output.

**Comment by totto82** [comment]
I have looked at the test failures and they are ok. i.e. from my point of view this can be merged after an update of reference data.

**Comment by totto82** [comment]
jenkins build this please

**Comment by totto82** [comment]
@blattms Can this be merged?

File: opm/simulators/wells/BlackoilWellModel_impl.hpp [hunk 2]
```diff
@@ -463,6 +465,34 @@ namespace Opm {
...
+
+            for (const auto& [name, to] : switched_prod_groups_) {
+                std::string msg = "    Production Group " + name
+                + " control mode changed from ";
+                const Group::ProductionCMode& oldControl = this->prevWGState().group_state.production_control(name);
+                std::string from = Group::ProductionCMode2String(oldControl);
+                msg += from;
+                msg += " to " + to;
+                if (to != from) {
+                    local_deferredLogger.info(msg);
```
Comment (blattms) [inline opm/simulators/wells/BlackoilWellModel_impl.hpp:479]: We could move the string construction to the `if (to != from)` clause

File: opm/simulators/wells/BlackoilWellModel_impl.hpp [hunk 3]
```diff
@@ -463,6 +465,34 @@ namespace Opm {
...
+                const std::string& name = key.first;
+                const Opm::Phase& phase = key.second;
+                std::string msg = "    Injection Group " + name
+                + " control mode changed from ";
+                const Group::InjectionCMode& oldControl = this->prevWGState().group_state.injection_control(name, phase);
+                std::string from = Group::InjectionCMode2String(oldControl);
+                msg += from;
+                msg += " to " + to;
+                if (to != from) {
+                    local_deferredLogger.info(msg);
```
Comment (blattms) [inline opm/simulators/wells/BlackoilWellModel_impl.hpp:492]: Same here. Only construct string if needed in if-clause

**Review by blattms — APPROVED** [review]
I think this can be merged. Just some comments that can be seen as nitpicking/cosmetical.

**Comment by totto82** [comment]
jenkins build this please

**Comment by totto82** [comment]
jenkins build this update_data please

**Comment by totto82** [comment]
jenkins build this opm-tests=802 please

**Comment by OPMUSER** [comment]
@totto82 
```
I suggest doing the same for the wells as
ORAT->GRAT->ORAT type of output may not be that useful for the user IMO.
```
I think this okay if we send the original message to the *.DBG file.

**Pull request merged**